ikarus/finiteelements/ferequirements.hh

```diff
@@ -561,6 +501,7 @@ public:
       DUNE_THROW(Dune::InvalidStateException, "This function can only be called when a single result is requested");
     }
   }
+  auto hasSingleResultRequested() const { return resType.size() == 1; }
```

Comment:
IMHO we now have a hybrid situation now. We have the bad things of both worlds. I suggest we go all-in and throw away Result requirements as well. Then, at the callsite we would have:

```cpp
auto feReq = FERequirements()...;

auto linearStress = fe.calculateAt(feReq, local, ResultType::linearStress);
```
Then the interface is superclear. 
You can also investigate in the posibility of doing it as:
```cpp
auto linearStress = fe.calculateAt<ResultType::linearStress>(feReq, local);
```
then the size of the result can be also decided at compile time and we do not have to use the funny type `Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, 0, 9, 3`

Reply:
Can you consider this and then I will continue with the review, or do you have urgent questions now?

Reply:
Yes I would also like that (especially the template). I think we would then need an additional argument for the global solution right?